scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -521,12 +525,13 @@ private Record(ProtocolVersion version, int epoch, long sequenceNumber) {
 		plaintext.writeBytes(getBlockCipherMac(writeState, compressedFragment));
 
 		// determine padding length
-		int ciphertextLength = compressedFragment.length + writeState.getCipherSuite().getMacLength() + 1;
-		int smallestMultipleOfBlocksize = writeState.getRecordIvLength();
+		int ciphertextLength = compressedFragment.length + writeState.getMacLength() + 1;
+		int blocksize = writeState.getRecordIvLength();
+		int smallestMultipleOfBlocksize = blocksize;
 		while ( smallestMultipleOfBlocksize <= ciphertextLength) {
-			smallestMultipleOfBlocksize += writeState.getRecordIvLength();
+			smallestMultipleOfBlocksize += blocksize;
 		}
-		int paddingLength = smallestMultipleOfBlocksize % ciphertextLength;
+		int paddingLength = smallestMultipleOfBlocksize - ciphertextLength;
```

Comment:
was it a bug ?

Reply:
No. The smallestMultipleOfBlocksize is just a little larger than ciphertextLength.

ctl = 103;
blockssize = 7 * 16 = 112

112 % 103 => 9
112 - 103 => 9

(I was surprised by this calculation. I will go to remove the loop, but later ...).  
e.g. 16*16